### PR TITLE
Update API deprecations for 2.17

### PR DESCRIPTION
**Pull request by gal-revach**
Still missing decision about the clusters API and how to handle them. Currently, they are on the deprecation section

### CODE REVIEW

File: docs/developer/overview-developer.md [hunk 1]
```diff
@@ -12,10 +12,7 @@ Developers can access Run:ai through various programmatic interfaces.
 
 The endpoints and parameters specified in the API reference are the ones that are officially supported by Run:ai. Endpoints and parameters that are **NOT** listed in the reference are not officially supported.
 
-Endpoints and parameters marked as `deprecated` means they remain operational and can be used; however, Run:ai will **NO LONGER RECOMMEND USING THEM**, **WILL NOT ADD FUNCTIONALITY TO THEM**, and **WILL NO LONGER BE SUPPORTED AFTER**
-
-Option1 : <#of versions>
-Option 2: <time period>.
+API endpoints and fields marked as `deprecated` means they remain operational and can be used; however, Run:ai will **NO LONGER RECOMMEND USING THEM**, **WILL NOT ADD FUNCTIONALITY TO THEM**, and **WILL STOP SUPPORTING THEM** after 2 major releases (for self-hosted deployments), which are 6 months of support for SaaS deployments.
```
Comment (jasonnovichRunAI) [inline docs/developer/overview-developer.md:15]:
```suggestion
API endpoints and fields marked as `deprecated` means they remain operational and can be used; however, Run:ai will **NO LONGER RECOMMEND USING THEM**, **WILL NOT ADD FUNCTIONALITY TO THEM**, and **WILL STOP SUPPORTING THEM** after 2 major releases (for self-hosted deployments), and after 6 months of support for SaaS deployments.
```

File: docs/home/whats-new-2-17.md [hunk 2]
```diff
@@ -128,30 +128,69 @@ Deprecated features will be available for **two** versions ahead of the notifica
...
-The following list of API endpoints have been marked for deprecation:
+The following list of API endpoints and fields that have been marked for deprecation:
 
-#### Jobs, events, pods API (replaced by workloads/pods/events)
+#### Jobs and Pods API
 
-| Deprecated endpoint | Replacement endpoint |
+| Deprecated | Replacement |
 | -- | -- |
 | https://app.run.ai/v1/k8s/clusters/{uuid}/jobs | https://app.run.ai/api/v1/workloads |
```
Comment (yaelFrank) [inline docs/home/whats-new-2-17.md:139]: The prefix https://apps.run.ai is only relevant to SaaS. We dont know the prefix in self hosted so it is better to write https://[CP URL] or to just omit it

File: docs/home/whats-new-2-17.md [hunk 3]
```diff
@@ -128,30 +128,69 @@ Deprecated features will be available for **two** versions ahead of the notifica
...
 | https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/count | https://app.run.ai/api/v1/workloads/count |
 | https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/{jobId}/pods | https://app.run.ai/api/v1/workloads/{workloadId}/pods |
 | https://app.run.ai/v1/k8s/clusters/{uuid}/pods | https://app.run.ai/api/v1/workloads/pods |
 
-#### Users, Applications, and Groups API
+#### Clusters API
+| Deprecated | Replacement |
+| -- | -- |
+| https://app.run.ai/v1/k8s/clusters | https://app.run.ai/api/v1/workloads |
+| https://app.run.ai/v1/k8s/clusters/{uuid} | https://app.run.ai/api/v1/workloads/count |
```
Comment (yaelFrank) [inline docs/home/whats-new-2-17.md:148]: something is all wrong here clusters are not replaced by workloads

File: docs/home/whats-new-2-17.md [hunk 4]
```diff
@@ -128,30 +128,69 @@ Deprecated features will be available for **two** versions ahead of the notifica
...
 | https://app.run.ai/v1/k8s/clusters/{uuid}/jobs | https://app.run.ai/api/v1/workloads |
 | https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/count | https://app.run.ai/api/v1/workloads/count |
 | https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/{jobId}/pods | https://app.run.ai/api/v1/workloads/{workloadId}/pods |
 | https://app.run.ai/v1/k8s/clusters/{uuid}/pods | https://app.run.ai/api/v1/workloads/pods |
 
-#### Users, Applications, and Groups API
+#### Clusters API
+| Deprecated | Replacement |
+| -- | -- |
+| https://app.run.ai/v1/k8s/clusters | https://app.run.ai/api/v1/workloads |
```
Comment (yaelFrank) [inline docs/home/whats-new-2-17.md:147]: something is all wrong here clusters are not replaced by workloads

**Review by yaelFrank — CHANGES_REQUESTED** [review]

File: docs/home/whats-new-2-17.md (hunk 2 above)
Comment (gal-revach) [reply docs/home/whats-new-2-17.md]: I removed the prefix what so ever, and left it as it is in the API docs

**Review by yaelFrank — APPROVED** [review]

**Comment by gal-revach** [comment]
Removed the cluster APIs - its breakage is from 2.16

File: docs/developer/overview-developer.md [hunk 5]
```diff
@@ -12,10 +12,7 @@ Developers can access Run:ai through various programmatic interfaces.
 
 The endpoints and parameters specified in the API reference are the ones that are officially supported by Run:ai. Endpoints and parameters that are **NOT** listed in the reference are not officially supported.
 
-Endpoints and parameters marked as `deprecated` means they remain operational and can be used; however, Run:ai will **NO LONGER RECOMMEND USING THEM**, **WILL NOT ADD FUNCTIONALITY TO THEM**, and **WILL NO LONGER BE SUPPORTED AFTER**
-
-Option1 : <#of versions>
-Option 2: <time period>.
+API endpoints and fields marked as `deprecated` means they remain operational and can be used; however, Run:ai will **NO LONGER RECOMMEND USING THEM**, **WILL NOT ADD FUNCTIONALITY TO THEM**, and **WILL STOP SUPPORTING THEM** after 2 major releases (for self-hosted deployments), and after 6 months of support for SaaS deployments.
```
[the remark this answers is not in the document]
Comment (jasonnovichRunAI) [reply docs/developer/overview-developer.md:15]:
```suggestion
API endpoints and fields marked as `deprecated` remain operational and can be used; however, Run:ai will **NO LONGER RECOMMEND USING THEM**, **WILL NOT ADD FUNCTIONALITY TO THEM**, and **WILL STOP SUPPORTING THEM** after 2 major releases (for self-hosted deployments), and after 6 months of support for SaaS deployments.
```

File: docs/home/whats-new-2-17.md [hunk 6]
```diff
@@ -128,30 +128,80 @@ Deprecated features will be available for **two** versions ahead of the notifica
...
+| /v1/k8s/clusters/{clusterId}/projects - _gpuOverQuotaWeight_ field | Use _nodePoolResources_ field |
+
+##### Departments API
+
+| Deprecated | Replacement |
+| -- | -- |
+| /v1/k8s/clusters/{clusterId}/projects - _resources_ field | Use _nodePoolResources_ field |
+| /v1/k8s/clusters/{clusterId}/projects - _deservedGpus_ field | Use _nodePoolResources_ field |
+| /v1/k8s/clusters/{clusterId}/projects - _allowOverQuota_ field | Use _nodePoolResources_ field |
+| /v1/k8s/clusters/{clusterId}/projects - _maxAllowedGpus_ field | Use _nodePoolResources_ field |
```
[the remark this answers is not in the document]
Comment (gal-revach) [reply docs/home/whats-new-2-17.md]: Fixed

File: docs/home/whats-new-2-17.md [hunk 7]
```diff
@@ -128,30 +128,100 @@ Deprecated features will be available for **two** versions ahead of the notifica
...
 
-| Deprecated endpoint | Replacement endpoint |
+| Endpoint | Change |
 | -- | -- |
-| https://app.run.ai/v1/k8s/clusters/{clusterUuid}/metrics | https://app.run.ai/api/v2/clusters/{clusterUuid}/metrics |
+| /api/v1/asset/compute | `gpuRequest` field was removed and is replaced by the following fields `gpuDevicesRequest`, `gpuRequestType`, `gpuPortionRequest`, `gpuPortionLimit`, `gpuMemoryRequest`, `gpuMemoryLimit`, `migProfile` |
+
+### Metrics deprecations
+
+The following metrics are deprecated and replaced by API. For details about the replacement APIs, see [Changed Metrics](../developer/metrics/metrics.md#changed-metrics-and-api-mapping):
```
Comment (jasonnovichRunAI) [inline docs/home/whats-new-2-17.md:208]:
```suggestion
The following metrics are deprecated and replaced by API endpoints. For details about the replacement APIs, see [Changed Metrics](../developer/metrics/metrics.md#changed-metrics-and-api-mapping):
```

**Review by jasonnovichRunAI — APPROVED** [review]

**Pull request merged**